modules/gapi/src/backends/onnx/dml_ep.cpp

```diff
@@ -13,13 +13,201 @@
 #ifdef HAVE_ONNX_DML
 #include "../providers/dml/dml_provider_factory.h"
 
+// FIXME It must be different #ifdef
+// E.g HAVE_DXCORE && HAVE_DIRECTX12 && HAVE_DIRECTML
+#ifdef HAVE_DXCORE
+
+// FIXME: Fix warning
+#define WINVER 0x0A00
```

Comment:
OpenCV defines `Windows 7` there:
https://github.com/opencv/opencv/blob/4.x/cmake/OpenCVCompilerOptions.cmake#L477-L478

Reply:
But `DXCore` minimum required version is `Windows 10`:

Taken from system header: `<dxcore.h>`
![image](https://github.com/opencv/opencv/assets/22346834/f9c9d793-5cb6-4347-8d57-8f250f18983a)

Reply:
1. I propose to set `WINVER` globally in CMake.
2. Cmake should print this during configuration.

Reply:
It's already defined in cmake: https://github.com/opencv/opencv/blob/4.x/cmake/platforms/OpenCV-WinRT.cmake#L18
This is how `videoio` handles that: https://github.com/opencv/opencv/blob/4.x/modules/videoio/src/cap_msmf.cpp#L13-L17